File: on promises/server/jtekt-server/src/main/java/fr/akensys/jtektserver/model/Mean.java
package fr.akensys.jtektserver.model;

import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Mean {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String storage;
    private String serial_number;
    private String licence_number;
    private String type;
    private String name;
    private State in_out;
    private LocalDateTime lastDate;
    private String meanNumber;
    
    @JsonManagedReference
    @OneToMany(mappedBy = "mean", cascade = CascadeType.ALL)
    private List<History> histories;
}

File: on promises/web/jtekt-client/src/app/models/mean.ts
export interface History {
  id: number;
  username: string;
  created_at: string;
  in_out: string;
  duration_out: number | null;
  duration_in: number | null;
}

export interface Mean {
  id: number;
  storage: number;
  serial_number: string;
  licence_number: string;
  type: string;
  name: string;
  in_out: string;
  lastDate: string;
  meanNumber: string;
  histories: History[];
}
